src/main/kotlin/controllers/MainController.kt

```diff
@@ -266,56 +294,36 @@ class MainController constructor(
 
             // capture the response
             val inputStream = conn.inputStream
-            val bufferedReader = BufferedReader(InputStreamReader(inputStream))
             val buffer = StringBuilder()
-
-            var line: String?
-            while (true) {
-                line = bufferedReader.readLine()
-                if (line != null) buffer.append(line)
-                else break
-            }
-            val data = buffer.toString()
-
-            // map the response to an array of GitHubRelease
-            val objectMapper = ObjectMapper()
-            val releases = objectMapper.readValue<Array<GitHubRelease>>(
-                data,
-                objectMapper.typeFactory.constructArrayType(GitHubRelease::class.java)
-            )
-
-            // determine if there is a newer version available
-            val currVersion = PackageMetadata.VERSION.split(".").map { it.toInt() }
-            var newerVersionUrl: String? = null
-
-            for (release in releases) {
-                val version = release.tagName.split(".").map { it.toInt() }
-
-                if (isVersionNewer(version, currVersion)) {
-                    newerVersionUrl = release.htmlURL
-                    break
+            BufferedReader(InputStreamReader(inputStream)).use { bufferedReader ->
+                var line: String?
+                while (true) {
+                    line = bufferedReader.readLine()
+                    if (line != null) buffer.append(line)
+                    else break
                 }
             }
-
-            if (newerVersionUrl != null) {
-                jMenuBar.add(
-                    JLinkLabel(
-                        newerVersionUrl,
-                        "Update available! Click here to download."
-                    )
-                )
-                jMenuBar.add(Box.createHorizontalStrut(4))
-                pack()
-            }
+            return buffer.toString()
         } catch (e: IOException) {
             e.printStackTrace()
+            return null
         }
+    }
 
-        configuration.saveProp(SettingsController.LAST_CHECKED_FOR_UPDATES_PROP, now.toString())
+    private fun newerReleaseExists(currVersion: List<Int>, releases: Array<GitHubRelease>): String? {
```

Comment:
Since this also returns the URL, might want a better name like `getUrlOfNewerRelease`.

BTW, if there are multiple newer versions, which one does it return?

Reply:
The results come back from the API sorted by date, so it should always pick the newest release that has a higher version number. However, this would undesired behavior if we release patches to older versions. I think to fix this, we should re-sort the list ourselves by tag.